api/test/e2enew/ssl/ssl_test.go

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ssl
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+
+	"e2enew/base"
+)
+
+var _ = ginkgo.Describe("SSL Basic", func() {
+	ginkgo.It("test ssl basic", func() {
+		// build test body
+		t := ginkgo.GinkgoT()
+		testCert, err := ioutil.ReadFile("../../certs/test2.crt")
+		assert.Nil(t, err)
+		testKey, err := ioutil.ReadFile("../../certs/test2.key")
+		assert.Nil(t, err)
+		apisixKey, err := ioutil.ReadFile("../../certs/apisix.key")
+		assert.Nil(t, err)
+		body, err := json.Marshal(map[string]interface{}{
+			"id":   "1",
+			"cert": string(testCert),
+			"key":  string(testKey),
+			"labels": map[string]string{
+				"build":   "16",
+				"env":     "production",
+				"version": "v3",
+			},
+		})
+		assert.Nil(t, err)
+		invalidBody, err := json.Marshal(map[string]string{
+			"id":   "1",
+			"cert": string(testCert),
+			"key":  string(apisixKey),
+		})
+		assert.Nil(t, err)
+		// Before configuring SSL, make a HTTPS request
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		http.DefaultTransport.(*http.Transport).DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+			if addr == "www.test2.com:9443" {
+				addr = "127.0.0.1:9443"
+			}
+			dialer := &net.Dialer{}
+			return dialer.DialContext(ctx, network, addr)
+		}
+		_, err = http.Get("https://www.test2.com:9443")
+		assert.NotNil(t, err)
+		assert.EqualError(t, err, "Get https://www.test2.com:9443: remote error: tls: internal error")
+		//create ssl fail - key and cert not match
+		base.RunTestCase(base.HttpTestCase{
```

Comment:
we could use table entry for these test cases.

Reply:
This use case requires the use of serialized SSL body incalidbody data.
Body data cannot be processed with table entry.
Do you have any good suggestions？

Reply:
oh, I see. ignore that.

Reply:
ok.